internal/scalibrextract/extract.go

```diff
@@ -103,19 +115,37 @@ func extractWithExtractor(ctx context.Context, localPath string, info fs.FileInf
 	return invsCompact, nil
 }
 
-func createScanInput(path string, fileInfo fs.FileInfo) (*filesystem.ScanInput, error) {
+func createScanInput(path string, root string, fileInfo fs.FileInfo) (*filesystem.ScanInput, error) {
 	reader, err := os.Open(path)
 	if err != nil {
 		return nil, err
 	}
 
+	// Rel will strip root from the input path.
+	path, err = filepath.Rel(root, path)
+	if err != nil {
+		return nil, err
+	}
+
 	si := filesystem.ScanInput{
-		FS:     os.DirFS("/").(scalibrfs.FS),
+		FS:     os.DirFS(root).(scalibrfs.FS),
 		Path:   path,
-		Root:   "/",
+		Root:   root,
 		Reader: reader,
 		Info:   fileInfo,
 	}
 
 	return &si, nil
 }
+
+func getRootDir(path string) string {
```

Comment:
can you add some comment to this function? I don't quite get the logic here (maybe due to lacking of context)

Reply:
Added some comments. This is just used to set the Root variable in scalibr. This is needed because go's fs.FS cannot use absolute paths, and always has to be relative to something. This gets the top level directory for it to be rooted at.

Reply:
I don't see new commits?

Reply:
Didn't push 🤦 , updated now

Reply:
thanks!